currency_rate_update_oxr/models/res_currency_rate_provider_OXR.py

```diff
@@ -63,7 +64,10 @@ def _obtain_rates(self, base_currency, currencies, date_from, date_to):
                     if 'description' in data
                     else 'Unknown error'
                 )
-            date_content = content[date.isoformat()]
+            data_date = datetime.utcfromtimestamp(data['timestamp']).date()
+            if eod_rates and data_date != date:
+                raise UserError(f"Exchange rate from incorrect date received. Got {data_date}, expected {date}")
```

Comment:
Because OCA aims at compatibility with the earliest Python version that the target Odoo version is compatible with, you cannot use f-strings in OCA/12.0 code.

Reply:
That's why I haven't seen it being used! Thanks for the catch.